node/src/neighborhood/node_record.rs

```diff
@@ -583,124 +620,100 @@ mod tests {
     #[test]
     fn node_record_partial_eq() {
         let earning_wallet = make_wallet("wallet");
+        let node_record_data = NodeRecordInputs {
+            earning_wallet: earning_wallet.clone(),
+            rate_pack: rate_pack(100),
+            accepts_connections: true,
+            routes_data: true,
+            version: 0,
+            location: None,
+        };
+        let node_record_data_duplicate = node_record_data.clone();
+        let node_record_data_with_neighbor = node_record_data.clone();
+        let node_record_data_mod_key = node_record_data.clone();
+        let mut node_record_data_mod_earning_wallet = node_record_data.clone();
+        let mut node_record_data_mod_rate_pack = node_record_data.clone();
+        let mut node_record_data_mod_accepts_connections = node_record_data.clone();
+        let mut node_record_data_mod_routes_data = node_record_data.clone();
+        let mut node_record_data_mod_version = node_record_data.clone();
+        let node_record_data_mod_signed_gossip = node_record_data.clone();
+        let node_record_data_mod_signature = node_record_data.clone();
+        let node_record_data_mod_node_addr = node_record_data.clone();
         let exemplar = NodeRecord::new(
             &PublicKey::new(&b"poke"[..]),
-            earning_wallet.clone(),
-            rate_pack(100),
-            true,
-            true,
-            0,
             main_cryptde(),
+            node_record_data,
         );
         let duplicate = NodeRecord::new(
             &PublicKey::new(&b"poke"[..]),
-            earning_wallet.clone(),
-            rate_pack(100),
-            true,
-            true,
-            0,
             main_cryptde(),
+            node_record_data_duplicate,
         );
         let mut with_neighbor = NodeRecord::new(
             &PublicKey::new(&b"poke"[..]),
-            earning_wallet.clone(),
-            rate_pack(100),
-            true,
-            true,
-            0,
             main_cryptde(),
+            node_record_data_with_neighbor,
         );
         let mod_key = NodeRecord::new(
             &PublicKey::new(&b"kope"[..]),
-            earning_wallet.clone(),
-            rate_pack(100),
-            true,
-            true,
-            0,
             main_cryptde(),
+            node_record_data_mod_key,
         );
         with_neighbor
```

Comment:
Can you please move this modification step towards the newly initialized variable (without having it intercepted by another initialization which makes it less bright)?